### DISCUSSION TITLE
is there a way to use anchorName/anchorPosition in stylex?

### DISCUSSION THREAD

**Discussion by lifeiscontent** [facebook/stylex#669] [Q&A] [answered]
I tried using the defineVars for anchorPosition/anchorName but it returns `var(--css-var)` instead of `--css-var` which is what they need as values

https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_anchor_positioning/Using

**Comment by nmn** [comment, marked as answer]
Using variables is supported for anchor positioning, but the value of the variable must start with `--`

e.g.

```ts
const anchors = stylex.defineVars({
  myAnchor: '--my-anchor',
});
```

The generated CSS will now be `var(--somehash)`, but it will resolve to `--my-anchor` and will work as expected.

You can also do this without `defineVars` at all and just use a string for `anchorName` directly:

```ts
anchorName: '--my-anchor',
```

This works and is explicitly supported.

---

We do not currently have a special case for these properties to avoid wrapping them in `var()`. This is intentional because you *can* use variables for those properties.

**Reply by nmn** [reply]
Note, support for `@position-try` was added recently as well.
